proposals/nnnn-streaming-media-manager-update.md

```diff
@@ -0,0 +1,148 @@
+
+The thought for implementation is as follows:
+```objc
+// Checking if SMM is paused
+if (streamingMediaManager.isVideoStreamingPaused) {
+	<#pause pulling of frames if running#>
+	return;
+}
+
+// Sending Video Data
+CVPixelBufferPoolRef pixelBufferPool = streamingMediaManager.pixelBufferPool;
+CGSize screenSize = streamingMediaManager.screenSize;
+
+CVImageBufferRef frame = <#acquire video frame#>
+[streamingMediaManager sendVideoFrame:frame];
+```
+
+##### Sending video data when iPhone is moving to background
+Since iOS cannot use the encoder when the phone is in the background, to conserve battery and CPU, we must find a solution that will drive the user to reopen the app. The suggested solution is to create a single frame that is generated when SMM’s encoder is initialized, and send that over as the Phone State changes from **Active** to **Resigning Active**. This should be built as a non-customizable screen, so that we can rely on consistent behavior among streaming applications for the time being. The current idea is to have a black screen with the text “Please re-open <#App Name#>”. We have the option to provide translations for all languages that SDL supports, if we wish to go that route. We can also use the app icon, if we wish instead of the app name, as the app name could potentially be quite long.
+
+##### Sending Audio Data
+Since audio data is sent sporadically, we will have a function that will send audio data only when necessary.
+```objc
+// Sending Audio Data
+[streamingMediaManager sendAudioData:<#audioData#>];
+```
+
+#### `SDLTouchManager`
+Currently, the `SDLTouchManager` class allows for developers to use a provided set of callbacks so they do not have to provide logic for determining gesture events such as tap, double tap, pinch, and pan. There are some developers, however, that would still like the ability to use their own. Currently, developers can listen to `SDLOnTouchEvent` notifications, and consume the object themselves, however a proposed solution is to make the internal `SDLTouch` object public, and allow for developers to use a handler on `SDLTouchManager` that will provide this touch object back, along with the `SDLTouchType`, to make it easier to handle and consume `SDLOnTouchEvent`.
+
+We would add a property to `SDLTouchManager` called `touchEventHandler`:
+```objc
+typedef void(^SDLTouchEventHandler)(SDLTouch *touch, SDLTouchType type);
+
+/**
+*  @abstract
+*      Returns all OnTouchEvent notifications as SDLTouch and SDLTouchType objects.
+*/
+@property (copy, nonatomic, nullable) SDLTouchEventHandler touchEventHandler;
+
+```
+
+## Impact on existing code
+The impact on SDL is that the previous version of `SDLStreamingMediaManager` and all code associated will no longer work. This will require a developer to adapt to using this new method of streaming, but will greatly improve the efficiency of Mobile Navigation Apps.
```

Comment:
This is not only additive. We are completely removing the ability to start/stop the sessions, because we want consistent lifecycle between all streaming apps.